File: javascript/leetcode-75/prefix-sum/find-the-highest-altitude.js
var largestAltitude = function(gain) {

    // prefix sum is a mathematics technique to total numbers in an array
    
    // we use this to check the max altitude of the totals against the previous max

    let maxAltitude = 0
    let currentAltitude = 0

    for (let i = 0; i < gain.length; i++) {

        // add numbers to currentAltitude as we iterate through the array
        currentAltitude += gain[i]

        // get max altitude
        maxAltitude = Math.max(maxAltitude, currentAltitude)
    }

    return maxAltitude
}

console.log(largestAltitude([-5,1,5,0,-7]))

console.log(largestAltitude([-4,-3,-2,-1,4,3,2]))

File: python/leetcode-75/prefix-sum/find-the-highest-altitude.py
from typing import List

def largestAltitude(gain: List[int]) -> int:
        current_altitude = 0
        max_altitude = 0

        for i in gain:
            current_altitude += i
            max_altitude = max(current_altitude, max_altitude)

        return max_altitude

print(largestAltitude([-5,1,5,0,-7]))

print(largestAltitude([-4,-3,-2,-1,4,3,2]))